### DISCUSSION TITLE
Configure Create3 and TurtleBot4 Raspberry Pi4 to prefer USB-C network?

### DISCUSSION THREAD

**Discussion by slowrunner** [turtlebot/turtlebot4#83] [Q&A] [answered]
It appears that the Create3 base of my TurtleBot 4 lite prefers to communicate with the TurtleBot 4 lite's Raspberry Pi4 via WiFi, and for some reason /battery_state topics stop passing to a client node some hours after the subscription is created.

I am suspicious that my Xfinity WiFi router somehow stops routing packets while maintaining the reserved DHCP connections, and quietly recovers routing.  I need to configure the Create3 and the TurtleBot4 (lite) Raspberry Pi4 to prefer the wired USB-C network to maintain reliable control if the WiFi router drops or drops packets.  (I have another Pi3B robot on the 2.4GHz WiFi network that reports periodic routing losses as well.)

Ref:  https://answers.ros.org/question/384008/ros2-prioritize-wifi-over-ethernet-by-default-whitelist-interface/

Interestingly, when the topics were not flowing to the client node (running on my desktop ROS2 Galactic environment), they magically started flowing again, when I stopped and started a /battery_state subscriber test node that was running on the TB4's RPi4.

**Comment by roni-kreinin** [comment, marked as answer]
@slowrunner We are working on getting the FastDDS Discovery Server working on TTB4 such that the Create 3 can be used without WiFi. See #26.
